spark/src/test/scala/org/apache/spark/sql/delta/OptimisticTransactionSuite.scala

```diff
@@ -806,4 +806,30 @@ class OptimisticTransactionSuite
       }
     }
   }
+
+  test("Skip computing state of post-commit snapshot") {
+    withTempDir { tableDir =>
+      val df = Seq((1, 0), (2, 1)).toDF("key", "value")
+      df.write.format("delta").mode("append").save(tableDir.getCanonicalPath)
+
+      val deltaLog = DeltaLog.forTable(spark, tableDir)
+      val snapshot = deltaLog.update()
+
+      assert(!snapshot.stateReconstructionTriggered)
+    }
+  }
+
+  test("Skip computing state of pre-commit snapshot") {
+    withTempDir { tableDir =>
+      val df = Seq((1, 0), (2, 1)).toDF("key", "value")
+      df.write.format("delta").mode("append").save(tableDir.getCanonicalPath)
+
+      val deltaLog = DeltaLog.forTable(spark, tableDir)
+      val preCommitSnapshot = deltaLog.update()
+
+      df.write.format("delta").mode("append").save(tableDir.getCanonicalPath)
+
+      assert(!preCommitSnapshot.stateReconstructionTriggered)
+    }
+  }
```

Comment:
I think we can easily combine these into a single test?
```suggestion
  test("Append does not trigger snapshot state computation") {
    withTempDir { tableDir =>
      val df = Seq((1, 0), (2, 1)).toDF("key", "value")
      df.write.format("delta").mode("append").save(tableDir.getCanonicalPath)

      val deltaLog = DeltaLog.forTable(spark, tableDir)
      val preCommitSnapshot = deltaLog.update()
      assert(!preCommitsnapshot.stateReconstructionTriggered)

      df.write.format("delta").mode("append").save(tableDir.getCanonicalPath)

      val postCommitSnapshot = deltaLog.update()
      assert(!preCommitsnapshot.stateReconstructionTriggered)
      assert(!postCommitSnapshot.stateReconstructionTriggered)
    }
  }
```